virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -148,45 +212,37 @@ def _setup_grid_neighbors(self) -> None:
         """
         self.data.grid.set_neighbours(distance=sqrt(self.data.grid.cell_area))
 
-    def get_community_by_key(self, key: int) -> AnimalCommunity:
-        """Function to return the AnimalCommunity present in a given grid square.
-
-        This function exists principally to provide a callable for AnimalCommunity.
-
-        Args:
-            key: The specific grid square integer key associated with the community.
-
-        Returns:
-            The AnimalCommunity object in that grid square.
-
-        """
-        return self.communities[key]
-
     def _initialize_communities(self, functional_groups: list[FunctionalGroup]) -> None:
-        """Initialize the animal communities.
+        """Initialize the animal communities by creating and populating animal cohorts.
 
         Args:
             functional_groups: The list of functional groups that will populate the
             model.
         """
+        # Initialize communities dictionary with cell IDs as keys and empty lists for
+        # cohorts
+        self.communities = {cell_id: list() for cell_id in self.data.grid.cell_id}
+
+        # Iterate over each cell and functional group to create and populate cohorts
+        for cell_id in self.data.grid.cell_id:
+            for functional_group in functional_groups:
+                # Calculate the number of individuals using Damuth's Law
+                individuals = damuths_law(
+                    functional_group.adult_mass, functional_group.damuths_law_terms
+                )
```

Comment:
I'm not entirely sure, but it seems to me this law is used already within the the `AnimalCohort` constructor. Not sure if for the same purpose, but just flagging it in case this is an error.

Reply:
Good catch!